_data/validator-libraries-modern.yml

```diff
@@ -197,6 +197,20 @@
       draft: [7, 6, 4]
       license: MIT
       last-updated: "2022-08-31"
+    - name: "@exodus/schemasafe"
+      url: https://github.com/ExodusMovement/schemasafe
+      notes: |
+        For Node.js and browsers, with security and speed being the main focus. Pre-compiles schemas to JS functions.
+        [Supports OpenAPI `discriminator`](https://github.com/ExodusMovement/schemasafe/blob/master/doc/Discriminator-support.md).
+        Default behavior is hardened with additional schema coherence validation.
+      date-draft: [2020-12, 2019-09]
+      draft: [7, 6, 4]
+      license: MIT
+      compliance:
+        config:
+          docs: https://github.com/ExodusMovement/schemasafe/blob/master/doc/Options.md
+          instructions: "disable schema coherence checks and `format` assertion by setting option `mode: 'spec'`"
```

Comment:
Does this mean that spec-compliance is not the default operating mode?

Reply:
Yes, by default, schemasafe refuses to compile schemas that are likely mistakes, e.g. unused keywords and unreachable checks.

E.g. `{ "type": "array", "maxLength": 10 }` will refuse to compile by default.

Reply:
As a rule of thumb, schemasafe checks that each keyword is consumed and affects validation in some way, otherwise it's treated as a potential mistake.

A separate, `'strong'` mode ensures validation the other way around – that each property of the graph is validated by some keyword. E.g. effectively that `unevaluatedProperties` is not missing anywhere in the schema.